modeltreeproxy.h

```diff
@@ -0,0 +1,85 @@
+/****************************************************************************
+**
+** Copyright (C) 2017 N7 Mobile sp. z o. o.
+** Contact: http://n7mobile.pl/Space
+**
+** This file is part of ASN.1/ACN Plugin for QtCreator.
+**
+** Plugin was developed under a programme and funded by
+** European Space Agency.
+**
+** This Plugin is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** This Plugin is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#pragma once
+
+#include "modeltree.h"
+#include "modeltreenode.h"
+
+namespace Asn1Acn {
+namespace Internal {
+
+class ModelTreeProxy
```

Comment:
This because ModelTree has got getters functions public and setters private, but accessible thorugh friendship construct for a class that is allowed to change it. To let the tests class use setters functions, I intorduced ModelTreeProxy class, which is now a friend of ModelTree . Classes that needs to access ModelTree private API inherits from the proxy, and performs the calls to ModelTree through ModelTreeProxy.

Reply:
I'd simply go with making setter public (they are in fact - anyone can access them if one wants, via proxy), but mkay ;)

Reply:
(especially that I hope that at some point whole ModelTree will disappear)